pallets/liquidity-pools-gateway/src/lib.rs

```diff
@@ -715,45 +760,69 @@ pub mod pallet {
 			weight_used
 		}
 
-		/// Retrieves the router stored for the provided domain and sends the
-		/// message, calculating and returning the required weight for these
-		/// operations in the `DispatchResultWithPostInfo`.
+		/// Retrieves the routers stored for the provided domain and sends the
+		/// message using each, calculating and returning the required weight
+		/// for these operations in the `DispatchResultWithPostInfo`.
 		fn process_message(
 			domain: Domain,
 			sender: T::AccountId,
 			message: T::Message,
 		) -> DispatchResultWithPostInfo {
 			let read_weight = T::DbWeight::get().reads(1);
 
-			let router = DomainRouters::<T>::get(domain).ok_or(DispatchErrorWithPostInfo {
-				post_info: PostDispatchInfo {
-					actual_weight: Some(read_weight),
-					pays_fee: Pays::Yes,
-				},
-				error: Error::<T>::RouterNotFound.into(),
-			})?;
-
-			let post_dispatch_info_fn =
-				|actual_weight: Option<Weight>, extra_weight: Weight| -> PostDispatchInfo {
-					PostDispatchInfo {
-						actual_weight: Some(Self::get_outbound_message_processing_weight(
-							actual_weight,
-							extra_weight,
-						)),
+			let routers =
+				DomainMultiRouters::<T>::get(domain).ok_or(DispatchErrorWithPostInfo {
+					post_info: PostDispatchInfo {
+						actual_weight: Some(read_weight),
 						pays_fee: Pays::Yes,
+					},
+					error: Error::<T>::MultiRouterNotFound.into(),
+				})?;
+
+			let mut post_dispatch_info = PostDispatchInfo {
+				actual_weight: None,
+				pays_fee: Pays::Yes,
+			};
+
+			for router in routers {
+				match router.send(sender.clone(), message.serialize()) {
+					Ok(dispatch_info) => Self::update_total_post_dispatch_info_weight(
+						&mut post_dispatch_info,
+						dispatch_info.actual_weight,
+						read_weight,
+					),
+					Err(e) => {
+						Self::update_total_post_dispatch_info_weight(
+							&mut post_dispatch_info,
+							e.post_info.actual_weight,
+							read_weight,
+						);
+
+						return Err(DispatchErrorWithPostInfo {
+							post_info: post_dispatch_info,
+							error: e.error,
+						});
 					}
```

Comment:
Also for that: 
* The first router must send the message
* The next routers must send a `MessageProof` of said message